test/dataloader/kv_test.exs

```diff
@@ -74,6 +74,17 @@ defmodule Dataloader.KVTest do
     refute_receive(:querying)
   end
 
+  test "pending_batches? is true when the cache is already warm", %{loader: loader} do
+    loader = Dataloader.put(loader, Test, :users, "ben", @data[:users] |> List.first)
+
+    loader = Dataloader.load(loader, Test, :users, "ben")
+    assert Dataloader.pending_batches?(loader)
```

Comment:
Hey folks! I could use some help understanding the design behind this PR. It seems to me that if the value is already put, `load` should be a no-op right? And then in that case `penidng_batches?` is false.

Reply:
Load should almost be a no op. The problem that this is solving is that when you have a data loader and you tell it to load id 5 (which is cached) then open an after_load in the resolver. In this case the on load won't fire correctly because run is never hit (due to pending batches being false) this makes the resolver end with unresolved resolvers

Reply:
Ah yes that's no good. I wonder if the solution then is editing how `on_load` works? It isn't that I object very much to the explicit `:pending?` key, it's more that I would like to be able to keep `Dataloader.pending_batches? #=> true` meaning that actual querying is going to happen. Which is to say, eager use of the cache vs lazy use of the cache.

Reply:
The challenge is how would onload know?

Reply:
The challenge is how would onload know?

Reply:
Well, `on_load` hands off to the `Absinthe.Middleware.Dataloader` middleware, which means the trick is solving this in the `call/2` callback. Fortunately this should be doable. Right now the callback is:

```
    def call(%{state: :unresolved} = resolution, {loader, callback}) do
      %{
        resolution
        | context: Map.put(resolution.context, :loader, loader),
          state: :suspended,
          middleware: [{__MODULE__, callback} | resolution.middleware]
      }
    end
```

What we're doing here is unconditionally suspending it. Instead though we should check to see if the value already exists. Oh shoot that isn't possible, because that's contained in the `callback` function. Hmmmm.

OK, one solution would be to have a `version` counter or similar inside dataloader. It gets incremented when there's stuff to load, it stays the same if the cache is used and there's nothing to load.

In `resolution` there exists the `loader` as it was PRIOR to the resolver. The counter in that could be compared to the `loader` value in the 2nd argument, which is the loader as returned by the resolver function.

Not sure if I like that scheme, I'll have to think about it more.

Reply:
Rgr. Let me know how you want to proceed